doc/decisions/database_structure.md

```diff
@@ -0,0 +1,78 @@
+# Database Structure
+
+The main goal of this document is to describe the database structure/layout especially in terms of relations between data entities.
+
+## Problem
+
+Semantically the plant information can be described as a hierarchy of entities. E.g., a plant is always a member on the variety level, while a variety is a part of species, species is a part of genus, genus is a part of subfamily, subfamily is a part of family. Please take a look at the diagram below. Each of these entities can have their own properties, e.g., a variety under a specie can have different height than the specie that it belongs to.
+
+The challenge is to define the structure of the database in a way that the data can be easily queried and updated.
+
+Another goal is to be able to define relations between entities, e.g., a plant of one variety doesn't want to be neighbor to plant of some other variety.
+
+```mermaid
+classDiagram
+
+    class Family:::abstract {
+    }
+    class Subfamily:::abstract {
+    }
+    class Genus:::abstract {
+    }
+    class Species {
+      +int height
+      +int width
+    }
+    class Variety {
+      +int height
+      +int width
+    }
+
+    Species <|-- Variety
+    Genus <|-- Species
+    Subfamily <|-- Genus
+    Family <|-- Subfamily
+
+```
+
+## Constraints
+
+## Assumptions
+
+## Considered Alternatives
+
+1. Inheritance feature of PSQL
+
+Inheritance cannot solve the challenge described above. I'll leave the main points from the [PSQL documentation](https://www.postgresql.org/docs/current/ddl-inherit.html) here:
+
+> Inheritance does not automatically propagate data from INSERT or COPY commands to other tables in the inheritance hierarchy.
+
+> All check constraints and not-null constraints on a parent table are automatically inherited by its children, unless explicitly specified otherwise with NO INHERIT clauses. Other types of constraints (unique, primary key, and foreign key constraints) are not inherited.
+
+> Table inheritance is typically established when the child table is created, using the INHERITS clause of the CREATE TABLE statement.
+
+So the inheritance is useful to deal with complex DDL structure on the startup, but will not help us to avoid bulk operations e.g. updating a column for every `variety` in the entire `genus`
+
+## Decision
+
+1. Hierarchy
+
+Hierarchy of entities should be implemented using foreign keys and table relationships instead of Inheritance feature of PSQL. In case of a conflict, the lowest entity level in the hierarchy is prioritized. E.g. when there is a height value in a species and in a variety, variety wins.
```

Comment:
Actually it would be best if you can give a few SQL commands how we would update and select attributes from the plants.

Reply:
I agree with you that we should explicitly document table relationships. Since this document is a decision document I would like to keep technical details out of it i.e. the better place would be [database_schemata.md](https://github.com/ElektraInitiative/PermaplanT/blob/db-docs/doc/architecture/database_schemata.md) in my opinion

Reply:
If the details are part of what is to be decided, they need to be mentioned. (If the details are not relevant for the decision, they don't need to be mentioned.)